### DISCUSSION TITLE
Creating an app with Quasar + Amplify Auth : PWA vs Android (Cordova)

### DISCUSSION THREAD

**Discussion by jimmyl-cgair** [quasarframework/quasar#10908] [CLI - Capacitor / Cordova modes]
Hi, 
I am coding a solution that used **AWS Amplify Auth** (_**Cognito** federated by **Google**_). On the Google side, I did set an “OAuth 2.0 Client ID” of type “Web application”.

Running the project on **PWA mode** (#> quasar dev), it works fine. 

This is how the OAuth is done: 

1.	App is loaded with “Connect” button.
2.	Click on “Connect” button launch auth action: Cognito then Google Authentification then return to the page who required to be connected. 

But, when I build an Android version, signed the apk and installed it on my phone, I got **“Application Error” ERR_CONNECTION_TIMED_OUT** . 

I think that I need to set an “OAuth 2.0 Client ID” of type “Android”. 

How will I manage that in Quasar? How do the code will make difference between PWA or Android mode?
This is new for me, I have never done that.

**Comment by ontwikkelfabriek** [comment]
@jimmyl-cgair 

> I think that I need to set an “OAuth 2.0 Client ID” of type “Android”.

You'll need a **plugin** for `AWS Cognito` to work on Cordova/Capacitor apps: 
https://ionic.io/docs/auth-connect/aws-cognito ( Cordova  & Capacitor ):

> How will I manage that in Quasar? How do the code will make difference between PWA or Android mode?

There's more then 1 way to do this but the most straight forward way is to use Quasar's `platform detection` helpers:
https://quasar.dev/options/platform-detection

**Reply by metalsadman** [reply]
you can also use [process.env.MODE](https://quasar.dev/quasar-cli/handling-process-env#import-based-on-process-env) detection to strip out code or conditional imports, as using platform specific plugins in your case android, will cause an error in other platform.

**Reply by ontwikkelfabriek** [reply]
@metalsadman 

Is the an example of using `process.env.MODE` to `strip out code` ( using Webpack I presume) ?
